rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java

```diff
@@ -293,16 +311,58 @@ void verifyLockSenderIsWhitelisted_whenAddressIsNotWhitelisted_shouldReturnFalse
         assertFalse(actualResult);
     }
 
-    private void saveInMemoryStorageOneOffWhiteListEntry() {
-        Coin maxTransferValue = Coin.COIN;
-        OneOffWhiteListEntry oneOffWhiteListEntry = new OneOffWhiteListEntry(btcAddress, maxTransferValue);
-        List<OneOffWhiteListEntry> oneOffWhiteListEntries = Collections.singletonList(oneOffWhiteListEntry);
-        Pair<List<OneOffWhiteListEntry>, Integer> pairValue = Pair.of(oneOffWhiteListEntries, 100);
+    @Test
+    void save_whenLockWhitelistIsNull_shouldReturnZeroEntries() {
+        whitelistSupport.save();
 
-        inMemoryStorage.safeSaveToRepository(
-            LOCK_ONE_OFF.getKey(),
-            pairValue,
-            BridgeSerializationUtils::serializeOneOffLockWhitelist
-        );
+        int actualSize = whitelistSupport.getLockWhitelistSize();
+        assertEquals(0, actualSize);
+        assertNull(whitelistSupport.getLockWhitelistEntryByIndex(0));
+        assertNull(whitelistSupport.getLockWhitelistEntryByIndex(1));
+    }
+
+    @Test
+    void save_whenRSKIP87IsNotActive_shouldSaveOneOffLockWhitelistAddress() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(false);
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, WhitelistCaller.AUTHORIZED.getRskAddress());
+        whitelistSupport.addOneOffLockWhitelistAddress(tx, btcAddress.toString(), BigInteger.TEN);
+
+        whitelistSupport.save();
+
+        int actualSize = whitelistSupport.getLockWhitelistSize();
+        Address actualBtcAddress = whitelistSupport.getLockWhitelistEntryByAddress(btcAddress.toString()).address();
+        assertEquals(1, actualSize);
+        assertEquals(btcAddress, actualBtcAddress);
+    }
+
+    @Test
+    void save_whenRSKIP87IsActive_shouldSaveUnlimitedLockWhitelistAddress() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(true);
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, WhitelistCaller.AUTHORIZED.getRskAddress());
+        whitelistSupport.addUnlimitedLockWhitelistAddress(tx, btcAddress.toString());
+
+        whitelistSupport.save();
+
+        int actualSize = whitelistSupport.getLockWhitelistSize();
+        Address actualBtcAddress = whitelistSupport.getLockWhitelistEntryByAddress(btcAddress.toString()).address();
```

Comment:
This test and the one above seem to be basically the same. There is no assertion of what was saved, it only changes the method you call in WhitelistSupport. But doesn't seem to have any relationship with the `save` method right?

Reply:
Actually, they have a direct relation with the `save` method. At first, they use `addUnlimitedLockWhitelistAddress` and `addOneOffLockWhitelistAddress` to add an address to be whitelisted, and then the methods save those addresses in storage with the `save` method.

In the assertion part, you can see we guarantee the address was saved by getting the `LockWhitelistSize` from the storage and getting the address we saved with the `getLockWhitelistEntryByAddress` method directly from the storage as well.